include/swift/Basic/LangOptions.h

```diff
@@ -41,6 +42,9 @@ namespace swift {
     /// Language features
     ///
 
+    /// \brief User-overridable language version to compile for.
+    version::Version EffectiveLanguageVersion = version::Version::getCurrentLanguageVersion();
```

Comment:
Ah, the Clang importer basically makes the scenario I'm describing unworkable, because there's only one and it has to pick a language mode. So, that means `EffectiveLanguageVersion` should stay in `LangOptions`. Thanks, @jrose-apple